pkg/autodiscovery/providers/utils.go

```diff
@@ -100,22 +100,23 @@ func buildTemplates(key string, checkNames []string, initConfigs, instances []in
 
 // extractTemplatesFromMap looks for autodiscovery configurations in a given map
 // (either docker labels or kubernetes annotations) and returns them if found.
-func extractTemplatesFromMap(key string, input map[string]string, prefix string) ([]integration.Config, error) {
+func extractTemplatesFromMap(key string, input map[string]string, prefix string) ([]integration.Config, []error) {
 	configs := make([]integration.Config, 0)
+	errors := make([]error, 0)
```

Comment:
nit: to avoid useless allocations, we can make it `var errors []error`. Same for configs.
[`nil` is a valid default value for go slices](https://github.com/golang/go/wiki/CodeReviewComments#declaring-empty-slices)